keras/layers/core.py

```diff
@@ -372,24 +372,14 @@ def _fix_unknown_dimension(self, input_shape, output_shape):
         return tuple(output_shape)
 
     def compute_output_shape(self, input_shape):
-        return (input_shape[0],) + self._fix_unknown_dimension(
-            input_shape[1:], self.target_shape)
+        if None in input_shape[1:]:
+            return (input_shape[0],) + tuple(s if s != -1 else None for s in self.target_shape)
```

Comment:
Please add a comment describing what this does / why do it, and make this line shorter.

Reply:
Done.